docker/amdvlk.Dockerfile

```diff
@@ -67,29 +64,49 @@ COPY docker/update-llpc.sh /vulkandriver/
 
 # Build LLPC.
 WORKDIR /vulkandriver/builds/ci-build
-RUN EXTRA_FLAGS="" \
+RUN EXTRA_COMPILER_FLAGS=() \
+    && EXTRA_LINKER_FLAGS=() \
+    && EXTRA_FLAGS=() \
+    && SANITIZERS=() \
     && if echo "$FEATURES" | grep -q "+gcc" ; then \
-         EXTRA_FLAGS="$EXTRA_FLAGS -DCMAKE_C_COMPILER=gcc"; \
-         EXTRA_FLAGS="$EXTRA_FLAGS -DCMAKE_CXX_COMPILER=g++"; \
+         EXTRA_FLAGS+=("-DCMAKE_C_COMPILER=gcc"); \
+         EXTRA_FLAGS+=("-DCMAKE_CXX_COMPILER=g++"); \
        fi \
     && if echo "$FEATURES" | grep -q "+clang" ; then \
-         EXTRA_FLAGS="$EXTRA_FLAGS -DCMAKE_C_COMPILER=clang"; \
-         EXTRA_FLAGS="$EXTRA_FLAGS -DCMAKE_CXX_COMPILER=clang++"; \
-         EXTRA_FLAGS="$EXTRA_FLAGS -DLLVM_USE_LINKER=lld"; \
-         EXTRA_FLAGS="$EXTRA_FLAGS -DCMAKE_EXE_LINKER_FLAGS=-fuse-ld=lld"; \
-         EXTRA_FLAGS="$EXTRA_FLAGS -DCMAKE_SHARED_LINKER_FLAGS=-fuse-ld=lld"; \
+         EXTRA_FLAGS+=("-DCMAKE_C_COMPILER=clang"); \
+         EXTRA_FLAGS+=("-DCMAKE_CXX_COMPILER=clang++"); \
+         EXTRA_FLAGS+=("-DLLVM_USE_LINKER=lld"); \
+         EXTRA_LINKER_FLAGS+=("-fuse-ld=lld"); \
        fi \
     && if echo "$FEATURES" | grep -q "+shadercache" ; then \
-         EXTRA_FLAGS="$EXTRA_FLAGS -DLLPC_ENABLE_SHADER_CACHE=1"; \
+         EXTRA_FLAGS+=("-DLLPC_ENABLE_SHADER_CACHE=1"); \
        fi \
-    && if echo "$FEATURES" | grep -q "+sanitizers" ; then \
-         EXTRA_FLAGS="$EXTRA_FLAGS -DXGL_USE_SANITIZER=Address;Undefined"; \
+    && if echo "$FEATURES" | grep -q "+asan" ; then \
+         SANITIZERS+=("Address"); \
          echo "export ASAN_OPTIONS=detect_leaks=0" >> /vulkandriver/env.sh; \
          echo "export LD_PRELOAD=$(clang -print-file-name=libclang_rt.asan-x86_64.so)" >> /vulkandriver/env.sh; \
+       fi \
+    && if echo "$FEATURES" | grep -q "+ubsan" ; then \
+         SANITIZERS+=("Undefined"); \
+       fi \
+    && if [ ${#SANITIZERS[@]} -ne 0 ]; then  \
+         SANITIZERS_SEPARATED_LIST="${SANITIZERS[@]}"; \
+         SANITIZERS_SEPARATED_LIST="${SANITIZERS_SEPARATED_LIST// /;}"; \
+         EXTRA_FLAGS+=("-DXGL_USE_SANITIZER='${SANITIZERS_SEPARATED_LIST}'"); \ 
        else \
-         EXTRA_FLAGS="$EXTRA_FLAGS -DXGL_BUILD_CACHE_CREATOR=ON"; \
+         EXTRA_FLAGS+=("-DXGL_BUILD_CACHE_CREATOR=ON"); \    
```

Comment:
I think we should always enable cache creator now. In the pass, this was disabled because of running into the disk space quota. But since we added a cleanup step, this stopped being an issue.